test/EtomicSwap.js

```diff
@@ -99,6 +116,95 @@ contract('EtomicSwap', function(accounts) {
         await this.swap.erc20Payment(...params).should.be.rejectedWith(EVMThrow);
     });
 
+    it('should allow to send ERC721 payment', async function () {
+        const lockTime = await currentEvmTime() + 1000;
+        const tokenId = 1; // Assuming token ID 1 is minted to accounts[0] in Erc721Token contract
+        const params = [
+            id,
+            accounts[1], // Receiver address
+            this.erc721token.address, // ERC721 token contract address
+            tokenId, // Token ID
+            secretHash,
+            lockTime
+        ];
+
+        // approve the swap contract to transfer the token on behalf of accounts[0]
+        await this.erc721token.approve(this.swap.address, tokenId, { from: accounts[0] });
+        // call the erc721Payment function to send the token to the swap contract
+        await this.swap.erc721Payment(...params, { from: accounts[0] }).should.be.fulfilled;
+
+        // Check the payment lockTime and state
+        const payment = await this.swap.payments(id);
+        assert.equal(payment[1].valueOf(), lockTime);
+        assert.equal(payment[2].valueOf(), PAYMENT_SENT);
+
+        // Check the ownership of the token
+        const tokenOwner = await this.erc721token.ownerOf(tokenId);
+        assert.equal(tokenOwner, this.swap.address);
+
+        // should not allow to send again
+        await this.swap.erc721Payment(...params, { from: accounts[0] }).should.rejectedWith(EVMThrow);
+    });
+
+    it('should allow to send ERC1155 payment', async function () {
+        const lockTime = await currentEvmTime() + 1000;
+        const tokenId = 1; // Token ID used in Erc1155Token contract
+        const amountToSend = 2; // Amount of tokens to send
+        const params = [
+            id,
+            amountToSend,
+            accounts[1], // Receiver address
+            this.erc1155token.address, // ERC1155 token contract address
+            tokenId,
+            secretHash,
+            lockTime
+        ];
+
+        // Approve the swap contract to manage all accounts[0]'s tokens
+        await this.erc1155token.setApprovalForAll(this.swap.address, true, { from: accounts[0] });
```

Comment:
Can approval requirement be avoided by handling HTLC creation logic in `onERC1155Received`?

Reply:
Approve was removed